CRM/Export/BAO/ExportProcessor.php

```diff
@@ -515,32 +515,22 @@ public function getValidLocationFields() {
    */
   public function getSqlColumnDefinition($field) {
     $fieldName = $this->getMungedFieldName($field);
-
-    // early exit for master_id, CRM-12100
-    // in the DB it is an ID, but in the export, we retrive the display_name of the master record
-    // also for current_employer, CRM-16939
-    if ($fieldName == 'master_id' || $fieldName == 'current_employer') {
-      return "$fieldName varchar(128)";
-    }
-
-    if (substr($fieldName, -11) == 'campaign_id') {
-      // CRM-14398
-      return "$fieldName varchar(128)";
-    }
-
     $queryFields = $this->getQueryFields();
-    $lookUp = ['prefix_id', 'suffix_id'];
+    // special case: civicirm_primary_id exports the contact id and is an index field
+    if ($fieldName == 'civicrm_primary_id') {
+      return "$fieldName varchar(16)";
```

Comment:
Why would we use `varchar` to store an integer?

Reply:
So this change will need to be tested on servers with significant data sets for performance - we have a good theory as to what will work well but it needs some concerted testing. It's also possible we can & should eliminate the temp table per my comment above

Reply:
@colemanw : in an export, in general, a civi field stored as an integer might actually end up getting a long string value (e.g. the gender field, which is what triggerred this PR).

Reply:
Ok but in the case of `civicrm_primary_id` we know the temp table is storing an integer, right?